db/db_impl/db_impl_open.cc

```diff
@@ -1400,7 +1403,7 @@ Status DBImpl::InitializeLogReader(
       immutable_db_options_.info_log, std::move(file_reader), reporter,
       true /*checksum*/, wal_number,
       immutable_db_options_.track_and_verify_wals, stop_replay_for_corruption,
-      min_wal_number, predecessor_wal_info));
+      min_wal_number, predecessor_wal_info, corrupted_wal_number));
```

Comment:
It's a bit unusual to pass this state tracking variable into a log Reader's constructor. It is in some sense a output argument, but there is no clear syntax for it. 

Can we instead track this in `Reporter` with an explicit API like `GetCorruptedWalNumber`?

Reply:
> Can we instead track this in Reporter

Agree - I decided to track the corrupted predecessor wal number in reporter instead. Fixed.